docs/tutorials/basic/data.md

```diff
@@ -125,37 +48,43 @@ for batch in data_iter:
     print([batch.data, batch.label, batch.pad])
 ```
 
-## Read CSV
-
-There is an iterator called `CSVIter` to read data batches from CSV files. We
-first dump `data` into a csv file, and then load the data.
+## Reading data from CSV Files
+MXNet provides [`CSVIter`](http://mxnet.io/api/python/io.html#mxnet.io.CSVIter)
+to read from CSV Files and can be used as below:
 
 ```python
+#lets save `data` into a csv file first and try reading it back
 np.savetxt('data.csv', data, delimiter=',')
 data_iter = mx.io.CSVIter(data_csv='data.csv', data_shape=(3,), batch_size=30)
 for batch in data_iter:
     print([batch.data, batch.pad])
 ```
 
-Note that we have not given a label file, then `batch.label` is empty here.
-
-## Read images
+## Custom Iterator
+When the built-in iterators do not suit your application,
+you can create a custom data iterator.
 
-<!-- TODO(mli) move notebooks here -->
-
-- [Read images](https://github.com/dmlc/mxnet-notebooks/blob/master/python/basic/image_io.ipynb)
-- [Advanced image reading](https://github.com/dmlc/mxnet-notebooks/blob/master/python/basic/advanced_img_io.ipynb)
-
-## Write your own data iterators
-
-Sometimes the provided iterators are not enough for some application. There are
-mainly two ways to develop a new iterator. One is creating from scratch: the
-following codes define an iterator that creates a given number of data batches
-through a data generator `data_gen`.
+An iterator in _MXNet_ should
+1. Implement `next()` in ``Python2`` or `__next()__` in ``Python3``,
+   returning a `DataBatch` or raising a `StopIteration` exception if at the end of the datastream.
+2. Implement the `reset()` method to restart reading from the beginning.
+3. Have a `provide_data` attribute, consisting of a list of `DataDesc` objects that store the name, shape, type and layout information of the data (more info [here](http://mxnet.io/api/python/io.html#mxnet.io.DataBatch)).
+4. Have a `provide_label` attribute consisting of a list of `DataDesc` objects that store the name, shape, type and layout information of the label.
+
+You can either create a iterator from scratch
+by defining an iterator or reuse existing iterators to create a new iterator.
```

Comment:
Suggest rewording as:
"When creating a new iterator, you can either start from scratch and define an iterator or reuse one of the existing iterators."